service/src/main/java/uk/gov/hmcts/reform/fpl/model/CaseData.java

```diff
@@ -194,6 +194,15 @@
     private final JudgeAndLegalAdvisor judgeAndLegalAdvisor;
     private final C2DocumentBundle temporaryC2Document;
     private final List<Element<C2DocumentBundle>> c2DocumentBundle;
+
+    @JsonIgnore
+    public C2DocumentBundle getLastC2DocumentBundle() {
+        return Optional.ofNullable(c2DocumentBundle)
+            .map(ElementUtils::unwrapElements)
+            .map(c2DocumentBundleList -> c2DocumentBundleList.get(c2DocumentBundleList.size() - 1))
```

Comment:
This could run into the issue where `c2DocumentBundleList.size() == 0` and therefore 
```java
c2DocumentBundleList.get(c2DocumentBundleList.size() - 1) 
```
will throw an IndexOutOfBoundsException.
If you want to keep it in the optional chain you could add a filter on the size (i.e. `!list.isEmpty()`) in the line above.